homeassistant/util/unit_system.py

```diff
@@ -259,6 +259,8 @@ def _deprecated_unit_system(value: str) -> str:
         ("gas", UnitOfVolume.CUBIC_FEET): UnitOfVolume.CUBIC_METERS,
         # Convert non-metric precipitation
         ("precipitation", UnitOfLength.INCHES): UnitOfLength.MILLIMETERS,
+        # Convert non-metric pressure
+        ("pressure", UnitOfPressure.PSI): UnitOfPressure.BAR,
```

Comment:
I *think* the preferred unit these days for tire pressure, oil pressure etc. is kPa, not bar. That said it really doesn't matter much.
We should convert `inHg` too, I'd suggest convert it to `hPa` since it's likely to be an atmospheric pressure.